### PR TITLE
Update Cargo.toml

### PR DESCRIPTION
**Pull request by Dylan-DPC**
Bumps the dependency to at least 4.1.3 as per the advisory

[Advisory](https://rustsec.org/advisories/RUSTSEC-2024-0344.html) 

(even though this crate may not be using the function mentioned in the advisory, it is better to bump it to ensure that there is no chance of other dependencies pulling it)

### CODE REVIEW

**Comment by str4d** [comment]
I'm somewhat "meh" on this change. We already pin the necessary version in the lockfile, which means all release binaries are fine:
https://github.com/str4d/rage/blob/8b0b65e98adbbbe9ed3336481fd748644a2509aa/Cargo.lock#L751-L753

By default, `cargo install rage` will use the latest dependency versions, so there's no issue there when using stable Rust, and for older Rusts they will most likely use `cargo install --locked` which will use the pinned MSRV-compatible dependencies from above which include the correct version.

The only case where this dependency change helps is when using the `age` crate as a library dependency. And in those cases, I generally prefer to stick to SemVer and defer to the downstream application for deciding what dependency versions to use. We don't pin dependencies when e.g. an update fixes compilation on a specific platform, because there might be other changes (like MSRV bumps) that the downstream cannot deal with.

Basically what this comes down to is, should security fixes be considered part of the "public API" of a dependency that the `age` crate uses?

**Comment by Dylan-DPC** [comment]
Not wrong. Didn't realise this is a binary while making a p so yeah that makes sense. Though you should still do it for a libary because you can't expect cargo to always pick the latest
